agent-inject/agent/annotations.go

```diff
@@ -526,6 +535,30 @@ func Init(pod *corev1.Pod, cfg AgentConfig) error {
 		pod.ObjectMeta.Annotations[AnnotationAgentDisableKeepAlives] = cfg.DisableKeepAlives
 	}
 
+	// validate JSON patches
+	if _, ok := pod.ObjectMeta.Annotations[AnnotationAgentJsonPatch]; ok {
+		// ignore empty string
+		if pod.ObjectMeta.Annotations[AnnotationAgentJsonPatch] == "" {
+			delete(pod.ObjectMeta.Annotations, AnnotationAgentJsonPatch)
+		} else {
+			_, err := jsonpatch.DecodePatch([]byte(pod.ObjectMeta.Annotations[AnnotationAgentJsonPatch]))
+			if err != nil {
+				return fmt.Errorf("error parsing JSON patch for annotation %s: %+v", AnnotationAgentJsonPatch, err)
```

Comment:
In a few places in this PR you've used `%v` or `%+v` for errors, but I think `%s` is more common when printing, and `%w` is preferred when wrapping (as in this case) - any particular reason for `%+v`?